examples/image_reconstruction.ipynb

```diff
@@ -0,0 +1,705 @@
+    "            return len(self.dataset)   \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "T125CnoLw_P2"
+   },
+   "outputs": [],
+   "source": [
+    "# producing the proper datasets and dataloaders for image reconstruction application, using the \"mydataset\" function.\n",
+    "trn_dataset_new = mydataset(trn_dataset)\n",
+    "val_dataset_new = mydataset(val_dataset)\n",
+    "test_dataset_new = mydataset(test_dataset)\n",
+    "\n",
+    "trn_dataloader = DataLoader(trn_dataset_new, batch_size=batch_size, shuffle=True)\n",
+    "val_dataloader = DataLoader(val_dataset_new, batch_size=batch_size, shuffle=False)\n",
+    "test_dataloader = DataLoader(test_dataset_new, batch_size=1, shuffle=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "nZEYIll7w_P2",
+    "outputId": "153cc8a5-c020-448c-e894-4f45d3d8f82e"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ConvAutoencoder(\n",
+      "  (conv1): Conv2d(1, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))\n",
+      "  (conv2): Conv2d(32, 4, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))\n",
+      "  (pool): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)\n",
+      "  (t_conv1): ConvTranspose2d(4, 32, kernel_size=(2, 2), stride=(2, 2))\n",
+      "  (t_conv2): ConvTranspose2d(32, 1, kernel_size=(2, 2), stride=(2, 2))\n",
+      ")\n"
+     ]
+    }
+   ],
+   "source": [
+    "class ConvAutoencoder(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super(ConvAutoencoder, self).__init__()\n",
+    "        \n",
+    "        #encoder\n",
+    "        self.conv1 = nn.Conv2d(1, 32, 3, padding=1)  \n",
+    "        self.conv2 = nn.Conv2d(32, 4, 3, padding=1)\n",
+    "        self.pool = nn.MaxPool2d(2, 2)\n",
+    "        \n",
+    "        #decoder\n",
+    "        self.t_conv1 = nn.ConvTranspose2d(4, 32, 2, stride=2)\n",
+    "        self.t_conv2 = nn.ConvTranspose2d(32, 1, 2, stride=2)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        \n",
+    "        x = torch.relu(self.conv1(x))\n",
+    "        x = self.pool(x)\n",
+    "        x = torch.relu(self.conv2(x))\n",
+    "        x = self.pool(x)  # compressed representation\n",
+    "        \n",
+    "        x = torch.relu(self.t_conv1(x))\n",
+    "        x = torch.sigmoid(self.t_conv2(x))\n",
+    "                \n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "network = ConvAutoencoder()\n",
+    "print(network)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "zva1C8Fxw_P3"
+   },
+   "source": [
+    "In order to interact with the optimization process, call backs are defined and saved in codes below, which saves the lastweights, best weights and the logs in row."
```

Comment:
```suggestion
    "In order to interact with the optimization process, callbacks are defined and saved in codes below, which saves the last weights, best weights and the logs in rows."
```
You should put a link on "callbacks" here leading to the documentation page: https://poutyne.org/callbacks.html

Reply:
Also, indentation of the code is not pep8.

Reply:
Maybe we could use an `experiment` here to should an example using this API?